tools/update-conda.py

```diff
@@ -0,0 +1,154 @@
+
+# Get source directory from command line arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("sourcedir", help="Source directory")
+options = parser.parse_args()
+
+platforms = {
+    "linux-64": "linux",
+    "linux-aarch64": "linux-aarch64",
+    "osx-64": "macos",
+    "osx-arm64": "macos-arm64",
+    #"win-64": "win",
+}
+pythons = ["3.9", "3.10", "3.11"]
+tags = ["", "-dev"]
+sources = ["src"]
+
+
+def write_env_file(env_file: Path, dependencies: list[str]) -> None:
+    env_file.write_text(
+        """name: sage
+channels:
+  - conda-forge
+  - nodefaults
+dependencies:
+"""
+        + "".join(f"  - {req}" + "\n" for req in dependencies)
+    )
+    print(f"Conda environment file written to {env_file}")
+
+
+def update_conda(source_dir: Path) -> None:
+    pyproject_toml = source_dir / "pyproject.toml"
+    if not pyproject_toml.exists():
+        print(f"pyproject.toml not found in {pyproject_toml}")
+        return
+
+    dependencies = get_dependencies(pyproject_toml)
+
+    for platform_key, platform_value in platforms.items():
+        for python in pythons:
+            for tag in tags:
+                # Pin Python version
+                pinned_dependencies = {
+                    f"python={python}" if dep == "python" else dep
+                    for dep in dependencies
+                }
+                if tag == "-dev":
+                    dev_dependencies = get_dev_dependencies(pyproject_toml)
+                    print(f"Adding dev dependencies: {dev_dependencies}")
+                    pinned_dependencies = pinned_dependencies.union(dev_dependencies)
+                pinned_dependencies = sorted(pinned_dependencies)
+
+                for src in sources:
+                    env_file = source_dir / src / f"environment{tag}-{python}.yml"
+                    write_env_file(env_file, pinned_dependencies)
+                    lock_file = (
+                        source_dir / src / f"environment{tag}-{python}-{platform_value}"
+                    )
+                    print(
+                        f"Updating lock file for {env_file} at {lock_file}", flush=True
+                    )
+                    subprocess.run(
+                        [
+                            "conda-lock",
+                            "--channel",
+                            "conda-forge",
+                            "--kind",
+                            "env",
+                            "--platform",
+                            platform_key,
+                            "--file",
+                            str(env_file),
+                            "--lockfile",
+                            str(lock_file),
+                            "--filename-template",
+                            str(lock_file),
+                        ]
+                    )
+
+
+def get_dependencies(pyproject_toml: Path) -> list[str]:
+    graystull_config = Configuration("sagemath")
```

Comment:
Is this intentional or a typo?

Reply:
No of course not ;-) Thanks for catching it!